report_qweb_txt/models/ir_actions_report_xml.py

```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+# © 2016 Akretion (Alexis de Lattre <alexis.delattre@akretion.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, fields, api
+
+
+class IrActionsReportXml(models.Model):
+    _inherit = 'ir.actions.report.xml'
+
+    report_type = fields.Selection(selection_add=[('qweb-txt', 'Text')])
+
+    @api.model
+    def render_report(self, res_ids, name, data):
+        if data.get('report_type') == 'qweb-txt':
+            return self.env['report'].get_html(res_ids, name, data=data), 'txt'
```

Comment:
Suggested TODO: Add a filename extension field, visible when report_type is qweb-txt or some similar mechanism to support different file extensions. Alternatively I suppose something like

(qweb-txt, 'Text') , (qweb-txt-csv, 'CSV') then

```python
def render_report(self, res_ids, name, data):
        if data.get('report_type').startswith('qweb-txt'):
            return self.env['report'].get_html(res_ids, name, data=data), data['report_type'].rsplit('-')[0]
```

Would mean adding for example ZPL extension is as simple as

```python
report_type = fields.Selection(selection_add=[('qweb-txt-zpl', 'ZPL')])

Reply:
On v10, there is the field print_report_name on ir.actions.report.xml (which has "replaced" the OCA module "report_custom_filename"), but it seems to only work on PDF files, cf this code https://github.com/odoo/odoo/blob/10.0/addons/report/controllers/main.py#L107
So the solution proposed by @gdgellatly seems the best to me.

Reply:
@alexis-via actually in v11 I ended up solving my problems differently by integrating directly into the controller.  It solves the encoding and filename issues but has not really been tested in production yet.

https://github.com/odoonz/odoonz-addons/tree/11.0/qweb_txt_report

Take a look and take what you want.